packages/commerce-sdk-react/src/hooks/useAccessToken.ts

```diff
@@ -10,17 +10,28 @@ import useConfig from './useConfig'
 
 const onClient = typeof window !== 'undefined'
 
+interface AccessToken {
+    token: string | null
+    getTokenWhenReady: () => Promise<string>
+}
+
 /**
  * Hook that returns the access token.
  *
  */
-const useAccessToken = (): string | null => {
-    if (onClient) {
-        const config = useConfig()
-        return useLocalStorage(`${config.siteId}_access_token`)
-    }
+const useAccessToken = (): AccessToken => {
+    const config = useConfig()
     const auth = useAuthContext()
-    return auth.get('access_token')
+
+    const token = onClient
+        ? useLocalStorage(`${config.siteId}_access_token`)
+        : auth.get('access_token')
+
+    // NOTE: auth.ready() is to be called later. If you call it immediately in this hook,
+    // it'll cause infinite re-renders during testing.
```

Comment:
Does it cause infinite re-renders in production as well? If it does, this is fine. If it doesn't, it feels like we should change the tests, rather than this.

Reply:
Today `auth.ready()` is being called from 2 places:

1. on CommerceAPIProvider mount, in a `useEffect`
2. all SCAPI requests are gated by auth.ready()

Reply:
> the biggest takeaway for me is this: auth.ready() is meant to be called later

🤔 Where/when you call `auth.ready()` in user land should not be concern! If there is a limitation, we should fix it.